### DISCUSSION TITLE
MenuBarItem with icon

### DISCUSSION THREAD

**Discussion by limeniye** [unoplatform/uno#13145] [Q&A]
I`am trying to do MenuBarItem with icon.
![image](https://github.com/unoplatform/uno/assets/69314237/f6d477f5-d855-4076-bbfb-b77ddba4d6ed)

But for MenuBarItem I don't have a Header or Icon property.
How can I resolve this problem?
Currently I'am using Expander, but that's not what I`am expecting for.

**Comment by limeniye** [comment]
Currently I'am in the following state:
1. I installed CommunityToolkit Behaviors
```XML
	<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
		<PackageReference Include="CommunityToolkit.WinUI.UI.Behaviors" />
	</ItemGroup>
	<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) != 'windows'">
		<PackageReference Include="Uno.CommunityToolkit.WinUI.UI.Behaviors" />
	</ItemGroup>
```
```XML
	<PropertyGroup>
		<CommunityToolkitVersion>7.1.2</CommunityToolkitVersion>
		<UnoCommunityToolkit>7.1.200-dev.6.gf657bb2e91</UnoCommunityToolkit>
	</PropertyGroup>		
```

2. I wrote custom ShowAttachedFlyoutBehavior
```C#
public class ShowAttachedFlyoutBehavior
	: BehaviorBase<FrameworkElement>
{
	protected override bool Initialize()
	{
		var result = ShowAttachedFlyout();
		return result;
	}

	protected override bool Uninitialize()
	{
		HideAttachedFlyout();
		return true;
	}

	private bool ShowAttachedFlyout()
	{
		if (AssociatedObject == null)
		{
			return false;
		}
		FlyoutBase.ShowAttachedFlyout(AssociatedObject);
		return true;
	}

	private void HideAttachedFlyout()
	{
		FlyoutBase.GetAttachedFlyout(AssociatedObject).Hide();
	}
}
```

And I'am trying to execute it with a following code:
```XML
<Grid x:Name="AddNewGrid" ColumnDefinitions="70,Auto,20" Height="{Binding ElementName=NavigationPanel, Path=ActualHeight}">
	<FlyoutBase.AttachedFlyout>
		<MenuFlyout>
			<MenuFlyoutItem Text="UDP"/>
			<MenuFlyoutItem Text="TCP"/>
		</MenuFlyout>
	</FlyoutBase.AttachedFlyout>
	<i:Interaction.Behaviors>
		<core:EventTriggerBehavior EventName="Tapped" SourceObject="{Binding ElementName=AddNewGrid}">
			<behaviors:ShowAttachedFlyoutBehavior/>
		</core:EventTriggerBehavior>
	</i:Interaction.Behaviors>
	<TextBlock Grid.Column="1" Text="New" VerticalAlignment="Center"/>
</Grid>
```

But on deploy I have the following error:
![image](https://github.com/unoplatform/uno/assets/69314237/43247a7f-87b0-4a59-9c97-09386c4179e0)

```
Exception thrown: 'System.InvalidOperationException' in System.Private.CoreLib.dll
The text associated with this error code could not be found.
```

If I will remove this line ```<behaviors:ShowAttachedFlyoutBehavior/>``` the page will displayed.

**Reply by jeromelaban** [reply]
The error you're showing looks like it is coming from the official WCT repository on Windows, specifically. Have you tried a more recent build to see if it helps?

**Reply by limeniye** [reply]
@jeromelaban, I created a simple repo for this issue:

https://github.com/limeniye/UnoCustomBehaviorParsingError

[UnoCustomBehaviorParsingError-main.zip](https://github.com/unoplatform/uno/files/12343147/UnoCustomBehaviorParsingError-main.zip)

**Reply by jeromelaban** [reply]
Thanks. Have you tried a more recent build of the community toolkit to see if it helps?

**Reply by limeniye** [reply]
The version ```<CommunityToolkitVersion>7.1.2</CommunityToolkitVersion>``` Its the last version and also it hasn't .dev versions.

https://www.nuget.org/packages/CommunityToolkit.WinUI.UI.Behaviors#versions-body-tab

**Reply by jeromelaban** [reply]
Thanks. Then there's an error to report to create on the community toolkit report.

**Reply by limeniye** [reply]
https://github.com/CommunityToolkit/WindowsCommunityToolkit/discussions/4927
